src/Contacts/PhoneNumber.m

```diff
@@ -144,13 +144,17 @@ + (PhoneNumber *)tryParsePhoneNumberFromUserSpecifiedText:(NSString *)text {
         }
     };
 
-    if ([sanitizedString hasPrefix:@"+"]) {
-        // If the text starts with "+", just parse it.
-        tryParsingWithCountryCode(sanitizedString, [self defaultRegionCode]);
+    tryParsingWithCountryCode(sanitizedString, [self defaultRegionCode]);
 
+    if ([sanitizedString hasPrefix:@"+"]) {
+        // If the text starts with "+", don't try prepending
+        // anything else.
         return result;
     }
 
+    // Try just adding "+" and parsing it.
+    tryParsingWithCountryCode([NSString stringWithFormat:@"+%@", sanitizedString], [self defaultRegionCode]);
```

Comment:
The key fix: We should always try parsing by adding "+" regardless of whether it starts with the calling code that corresponds to the device's default region.